test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -795,6 +795,53 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
 		framework.ExpectNoError(disableAutoprovisioning())
 	})
+
+	It("shouldn't add new node group if not needed [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+		By("Decrease size of groups")
+		nodesCountAfterResize := 0
+		for mig, size := range originalSizes {
+			if size > 1 {
+				framework.ExpectNoError(framework.ResizeGroup(mig, int32(1)))
+				nodesCountAfterResize++
+			}
+		}
+		framework.ExpectNoError(framework.WaitForReadyNodes(c, nodesCountAfterResize, resizeTimeout))
+		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
```

Comment:
By downsizing existing node groups we will likely displace some system pods. Scheduler will attempt to put them on existing nodes, and if it fails to do so, CA will scale up the cluster. Meantime, rescheduler may place taints on nodes to reserve space for critical pods (for example Heapster.) So depending on which (if any) pods become unschedulable, this has a very large chance of resulting in different output from run to run.

If this test isn't intended to simulate node failure scenario, but just to verify a new node group is not created when not needed, it would be safer to ensure there is space in one of the groups by either:
- increasing max node count if necessary,
- creating a new, autoscaled node-pool with max size > initial size.

Reply:
Changed.